### PR TITLE
Add alarms for Sierra code description tables

### PR DESCRIPTION
**Pull request by aaronfriedman6**
Test that all the codes are transferred from Sierra, that there are no duplicates, and that all additional fields have been inferred or otherwise manually populated.

### CODE REVIEW

File: query_helper.py [hunk 1]
```diff
@@ -44,6 +50,38 @@
     "SELECT COUNT(key) FROM {table} "
     "WHERE transaction_et = '{date}';")
 
+_REDSHIFT_CODE_COUNT_QUERY = \
+    'SELECT COUNT({code}) FROM {table} WHERE deletion_date IS NULL;'
+
+_REDSHIFT_DISTINCT_CODE_COUNT_QUERY = \
+    'SELECT COUNT(DISTINCT {code}) FROM {table} WHERE deletion_date IS NULL;'
```
Comment (mwbenowitz) [inline query_helper.py:57]: Almost certainly an over-optimization, but these could be combined into a single query. If they're expensive it might help?

**Review by mwbenowitz — APPROVED** [review]
This looks good! I left a single comment about a potential optimization.

I also think that the three new alarm functions cold be abstracted a bit to share more functionality, but refactoring that would only be worth it if we're going to be adding more Sierra code tables in the future.

That's maybe something more to keep in mind for the future, meanwhile 👍

File: query_helper.py (hunk 1 above)
Comment (aaronfriedman6) [reply query_helper.py]: Good point -- done!

**Comment by aaronfriedman6** [comment]
> I also think that the three new alarm functions cold be abstracted a bit to share more functionality, but refactoring that would only be worth it if we're going to be adding more Sierra code tables in the future.


Yeah I thought the same thing but I believe these are going to be the only code tables with alarms. Eventually it might be worth it to totally refactor the alarm function into multiple classes etc. but I think that's a slightly bigger project than I want to take on right now.

**Pull request merged**